internal/gatewayapi/listener.go

```diff
@@ -112,6 +112,12 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 						EscapedSlashesAction: ir.UnescapeAndRedirect,
 					},
 				}
+				if len(listener.frontendValidationCACerts) > 0 {
```

Comment:
can we move this inside https://github.com/envoyproxy/gateway/blob/828edfb1cf320fd56002e29ddc01aa639cf71265/internal/gatewayapi/helpers.go#L390 ?